### ISSUE TITLE
Scope stretching various Code-Blocks

### ISSUE THREAD

**Issue by joehannes-zz** [eslint/eslint-plugin-markdown#74]
I am just finding my love for literate programming.
When coding and documenting heavily though, one thing doesn't work as _desired_ ...

* variable defined in one codeblock are not visible in other codeblocks

thus I can't use literate programming/markdown properly

* Is it me, not knowing best practices of literate programming
* Is there a workaround
* Could this be a feature request

I understand this is almost impossible to implement, unless you
* implement it for every literate programming lib out there, like write, literate-programming-lib, ...
* invent your own convention like a sub-header-level 2 signifies it's own scope or something
* simply think of one js file per md

However, I'm curious if you have thought about this and if there's a common best practice / workaround

**Comment by btmills** [comment]
Literate programming is definitely something I want to support with this plugin. It would most likely work by combining all JS code blocks into a single consecutive scope, basically just stripping out the Markdown to leave the JS. I haven't used that style myself, so can you point me to some other tools that already support it? I could add a setting in `.eslintrc.js` for just this plugin, but if there's a de facto standard for how to configure that, I'll use that instead.

**Comment by joehannes** [comment]
hey, sounds awesome!!!
Basically I use [joystylrs' lib](https://github.com/jostylr/literate-programming).
There's command line tools as well, it's called `litpro`, it's on npm.

For my latest project where I've been using all of that ... you can have a look at it [here](http://www.github.com/joehannes-jobs/webtrekk)
Just go to the literature folder and you'll find 4 markdown files which make up all of html, css, and js.

Litpro by joystylr is using conventions you could easily parse.
Basically he's using his `compile directives` like so `[path/to/be/compiled/and/put/there.js](#Section-Heading "save:")` ...

Now a ##Section Heading will be the starting point for this file ... until a new compile directive's heading begins.

I know there's other literate programming libs out there. I don't really know them, I just found this one to be pretty decent :) (eg: writ)

**Comment by jorgecasar** [comment]
"combining all JS code blocks into a single consecutive scope" it's a great feature to be able to validate markdown documentation using [Markdown javascript](https://open-wc.org/docs/experimental/mdjs/)

**Comment by btmills** [comment]
Based on what I've learned building this processor, it would be difficult to both extract code blocks individually and combine them as in literate programming. I'm closing this issue, but I'd welcome someone using this plugin as inspiration for building another plugin designed for combining code blocks. If you do build it, please share here!
